### DISCUSSION TITLE
.fetch_message() - Documentation

### DISCUSSION THREAD

**Discussion by iiLearner** [Rapptz/discord.py#6382] [Ideas]
Hello,

I could be very well wrong but there is something I noticed the method `.fetch_message()` is supported under `discord.Member` and `discord.User` but after tries I've made it does not seem to be working with these two. actually throws an exception.

However, it works perfectly for `discord.TextChannel`.
I'm sorry if im wrong, im pretty new to this and could possibly be wrong.

Good day.

**Comment by bandwiches** [comment]
Got any examples to provide?

**Comment by tk421bsod** [comment]
If you're using `get_member` to get a Member instance, but you're getting an error like `AttributeError: NoneType object has no attribute 'fetch_message'`, you'll need to enable the members intent in both the Discord Dev Portal and your code. Look at the [docs for intents](https://discordpy.readthedocs.io/en/latest/intents.html) for more information.

**Reply by iiLearner** [reply]
They are enabled @TK421bsod. This has nothing to do with it.

**Reply by bandwiches** [reply]
So then what exception are you getting?  [discord.Member.fetch_message()](https://discordpy.readthedocs.io/en/latest/api.html#discord.Member.fetch_message) throws 3 possible exceptions.  We can't help you without any details.

**Reply by iiLearner** [reply]
`AttributeError: 'ClientUser' object has no attribute 'fetch_message'`

`AttributeError: 'ClientUser' object has no attribute 'create_dm'` on Member.fetch_message

**Reply by tk421bsod** [reply]
You can't call fetch_message on a ClientUser, and you can't create a DM with yourself.

**Reply by Rapptz** [reply]
`Client.user`/`Bot.user` is a `ClientUser` and does not have those operations enabled.
